operator/v1/types_ingress.go

```diff
@@ -225,6 +232,97 @@ var (
 	DNSReadyIngressConditionType = "DNSReady"
 )
 
+// SecuritySpec defines the settings for securing IngressController connections.
+type SecuritySpec struct {
+	// Profile is one of three possible security configurations from [1]:
+	//
+	// [1] https://wiki.mozilla.org/Security/Server_Side_TLS#Recommended_configurations
+	//
+	// If unset, the default is "Intermediate".
+	//
+	// +optional
+	Profile *SecurityProfileType `json:"profile,omitempty"`
+	// CustomSettings defines the schema for customizing the security settings of
+	// IngressController connections.
+	//
+	// +optional
+	CustomSettings *CustomProfileSettings `json:"customSettings,omitempty"`
+}
+
+// SecurityProfileType defines a security profile.
+type SecurityProfileType string
+
+const (
+	// SecurityProfileOldType maps to:
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Old_backward_compatibility
+	SecurityProfileOldType SecurityProfileType = "Old"
+	// SecurityProfileIntermediateType maps to:
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Intermediate_compatibility_.28default.29
+	SecurityProfileIntermediateType SecurityProfileType = "Intermediate"
+	// SecurityProfileModernType maps to:
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Modern_compatibility
+	SecurityProfileModernType SecurityProfileType = "Modern"
+)
+
+// CustomProfileSettings defines the schema for a custom security profile.
+type CustomProfileSettings struct {
+	// Ciphers is used to specify the cipher algorithms that are negotiated
+	// during the SSL/TLS handshake with an IngressController. Each cipher must
+	// be an explicit, colon-delimited list of ciphers.
+	//
+	// If unset, the "Intermediate" ciphersuite [1] is used:
+	//
+	// [1] https://wiki.mozilla.org/Security/Server_Side_TLS#Intermediate_compatibility_.28default.29
+	//
+	// +optional
+	Ciphers *string `json:"ciphers,omitempty"`
```

Comment:
We should define an interaction with the base profile?  

i.e. either let the operator to know what the different levels mean and include the appropriate list for the level with anything set by this field as a concatenation?  That way you could say Intermediate as a base and set `!DES-CBC3-SHA` here if that cipher had some concerning attacks available.  You could specify `-ALL:` at the start of the cipher string if you wanted to replace the default with your explicit string.

OR you should say that this completely replaces the base `Profile`.

Reply:
As we discussed earlier today, `CustomSettings` fields will only be available for a `Profile` of type "Custom". All other `Profile` types are not configurable and individual settings can not be overridden.